### PR TITLE
Fix #80: Added a check on current server's php.ini's 'max_execution_time' requiring it to exceed the time specified / tested application's comfortable installation time

### PR DESCRIPTION
**Pull request by rossaddison**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

This would be a useful check prior to installation preventing unnecessary installation timeouts.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "yiisoft/requirements",
+    "name": "rossaddison/requirements",
```
Comment (samdark) [inline composer.json:2]:
```suggestion
    "name": "yiisoft/requirements",
```

File: src/RequirementsChecker.php [hunk 2]
```diff
@@ -294,6 +294,26 @@ public function checkUploadMaxFileSize(?string $min = null, ?string $max = null)
         return ($minCheckResult && $maxCheckResult);
     }
 
+    
+    /**
+     * Checks if the php.ini setting 'max_execution_time' exceeds a reliable and tested execution time 
+     * @param string|null $max
+     * @return bool
```
Comment (samdark) [inline src/RequirementsChecker.php:301]:
```suggestion
     * Checks if the `php.ini` setting `max_execution_time` exceeds the execution time specified.
     * @param string|null $max Maximum execution time.
     * @return bool True on success.
```

File: src/RequirementsChecker.php [hunk 3]
```diff
@@ -294,6 +294,26 @@ public function checkUploadMaxFileSize(?string $min = null, ?string $max = null)
...
+     */
+    public function checkMaxExecutionTime(?string $max = null): bool
```
Comment (samdark) [inline src/RequirementsChecker.php:303]: Why string if the time is int?

File: src/RequirementsChecker.php [hunk 4]
```diff
@@ -294,6 +294,26 @@ public function checkUploadMaxFileSize(?string $min = null, ?string $max = null)
...
+        $maxExecutionTime = ini_get('max_exection_time');
+        if ($max !== null) {
+            if ($maxExecutionTime < $max) {
+                $checkResult = false;
+            } else {
+                $checkResult = true;
+            }
+            return $checkResult;
+        }
+        return false;
```
Comment (samdark) [inline src/RequirementsChecker.php:314]:
```suggestion
        return $max === null || $maxExecutionTime <= $max;
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/RequirementsChecker.php (hunk 3 above)
Comment (rossaddison) [reply src/RequirementsChecker.php:303]: Just following the above function checkUploadMaxFileSize which uses strings for min and max. Yes should be int.

File: src/RequirementsChecker.php (hunk 4 above)
Comment (rossaddison) [reply src/RequirementsChecker.php:314]: Did a few test checks:

````
<?php
    function testServerLessThanInstallation(int $installationTimed = null) : bool {
        $maxExecutionTimed = 60;
        if ($installationTimed !== null) {
            if ($maxExecutionTimed < $installationTimed) {
                $checkResult = true;
            } else {
                $checkResult = false;
            }
        }
        return $checkResult;
    }
    echo testServerLessThanInstallation(240) ? 'failRequirements' : 'passRequirements';
?>
<br>
<?php 
    function testMaxExecutionTimeLessThanInstallationTime(int $applicationsInstallationTime = null) : bool {
       $maximumExecutionTime = 60;
       return ($applicationsInstallationTime === null || $maximumExecutionTime <= $applicationsInstallationTime);
    };
    echo testMaxExecutionTimeLessThanInstallationTime(240) ? 'failRequirements' : 'passRequirements';   
?>
````
The lengthier version and the shorter version yield the same results.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
All good. Please add a line for CHANGELOG.

**Comment by samdark** [comment]
Thank you!

**Pull request merged**